File: 78.子集.py
#
# @lc app=leetcode.cn id=78 lang=python3
#
# [78] 子集
#

# @lc code=start
from typing import List
from itertools import combinations
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        return [j for i in range(len(nums)+1) for j in combinations(nums,i)]
# @lc code=end

print(Solution().subsets([1,2,3]))

File: 78.子集.cpp
/*
 * @lc app=leetcode.cn id=78 lang=cpp
 *
 * [78] 子集
 */

// @lc code=start
#include<vector>
using namespace std;
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ret(1<<nums.size());
        for(int i=0; i<ret.size(); ++i){
            int tmp = i, idx = 0;
            while(tmp){
                if(tmp & 1) ret[i].push_back(nums[idx]);
                tmp >>= 1;
                ++idx;
            }
        }
        return ret;
    }
};
// @lc code=end

